### PR TITLE
Cookbook-Enums

### PR DESCRIPTION
**Pull request by idalithb**
-Needs review
-Missing page in other languages

### CODE REVIEW

**Review by schmidsi — CHANGES_REQUESTED** [review]
Great start. I left some comments to make this a bit clearer.

File: website/pages/en/cookbook/enums.mdx [hunk 1]
```diff
@@ -0,0 +1,968 @@
+---
+title: Optomize Subgraphs Using Enums
```
[the remark this answers is not in the document]
Comment (idalithb) [reply website/pages/en/cookbook/enums.mdx:2]: Yes, agree!

File: website/pages/en/cookbook/enums.mdx [hunk 2]
```diff
@@ -0,0 +1,968 @@
...
+---
+
+Learn how to effectively extract detailed information from transactions and marketplace interactions using Enums.
```
[the remark this answers is not in the document]
Comment (idalithb) [reply website/pages/en/cookbook/enums.mdx:5]: Great catch!

**Comment by chidubemokeke** [comment]
Looking good, thanks a ton for the support!

**Review by chidubemokeke — APPROVED** [review]
Love the edits, super thankful.

**Review by chidubemokeke — APPROVED** [review]

**Comment by idalithb** [comment]
> Love the edits, super thankful.

**Comment by idalithb** [comment]
> Love the edits, super thankful.

Wonderful! Happy to collab anytime! :)

**Comment by chidubemokeke** [comment]
> > Love the edits, super thankful.
> 
> @chidubemokeke Wonderful! Happy to collab anytime! :)

It's soothing to see new contributions from a different perspective. 

Super thankful to @schmidsi for the referral

**Review by chidubemokeke — APPROVED** [review]

File: website/pages/en/cookbook/enums.mdx [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+}
+```
+
+## Best Practices for Using Enums
+
+- **Consistent Naming:** Use clear, descriptive names for enum values to improve readability.
+- **Centralized Management:** Keep enums in a single file for easier updates and management.
+- **Documentation:** Add comments to enum definitions for better context.
+
+## Sample Queries
```
Comment (schmidsi) [inline website/pages/en/cookbook/enums.mdx:84]: I don't get the context of Enums in these queries. Either remove them completely or explain in detail some specifics of enums in queries:
1. Having enums as query filters
2. Having enums as part of the query response

Comment (idalithb) [reply website/pages/en/cookbook/enums.mdx:84]: @chidubemokeke which one would you prefer?

Comment (chidubemokeke) [reply website/pages/en/cookbook/enums.mdx:84]: @schmidsi  @idalithb  I updated the schema to improve queries for the enums. 

 Can you confirm?

Comment (chidubemokeke) [reply website/pages/en/cookbook/enums.mdx:84]: @schmidsi  I also updated the readme to give each query context in detail.

Comment (idalithb) [reply website/pages/en/cookbook/enums.mdx]: @chidubemokeke Updated to include your changes

**Review by chidubemokeke — APPROVED** [review]

**Review by chidubemokeke — APPROVED** [review]

**Review by chidubemokeke — APPROVED** [review]

File: website/pages/en/cookbook/enums.mdx [hunk 4]
```diff
@@ -0,0 +1,245 @@
+---
+title: Categorize NFT Marketplaces Using Enums
+---
+
+Learn how to effectively organize information from transactions and marketplace interactions using Enums.
```
Comment (alinobrasil) [inline website/pages/en/cookbook/enums.mdx:5]: The first sentence is the most important. I think the benefits should be summarized here so when someone considers reading, they see the end result first. For Enums, I'd say it helps make your code cleaner and less error-prone. I suggest something like: 
"Make your code cleaner and less error-prone by using Enums. Here's a full example of using enums on NFT marketplaces."

Generally, "learn" isn't an outcome people care about so I avoid it.

File: website/pages/en/cookbook/enums.mdx [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+
+> Note: This guide uses the CryptoCoven NFT smart contract.
+
+## What are Enums?
+
+Enums, or enumeration types, are a specific data type that allows you to define a set of allowed values.
```
Comment (alinobrasil) [inline website/pages/en/cookbook/enums.mdx:11]: I find an explanation without an example hard to follow. 

I suggest throwing in an example immediately before moving on to a new section. Perhaps no need for the `Enums Symtax` header below. Instead:
-Explain enum
-Show example right away
-Further explanations (the other content from Enums Syntax).

File: website/pages/en/cookbook/enums.mdx [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+
+This means that when you use the type `TokenStatus` in your schema, you expect it to be exactly one of `OriginalOwner`, `SecondOwner`, or `ThirdOwner`.
+
+To learn more about enums, check out [Creating a Subgraph](/developing/creating-a-subgraph/#enums) and [GraphQL documentation](https://graphql.org/learn/schema/#enumeration-types).
+
+## Benefits of Using Enums
+
+- **Clarity:** Enums provide meaningful names for values, making data easier to understand.
+- **Validation:** Enums enforce strict value definitions, preventing invalid data entries.
+- **Maintainability:** When you need to change or add new categories, enums allow you to do this in a focused manner.
```
Comment (alinobrasil) [inline website/pages/en/cookbook/enums.mdx:35]: How about adding a before & after comparison to highlight these benefits?

eg:
Before using enums, your code could look like this ...
After using enums, it's like this so typos won't affect you and the code is easier to maintain.

**Review by alinobrasil — COMMENTED** [review]
Here are a few edit suggestions, mostly in the beginning.

File: website/pages/en/cookbook/enums.mdx (hunk 4 above)
Comment (idalithb) [reply website/pages/en/cookbook/enums.mdx:5]: @alinobrasil Great point! This was a strategy I adopted from Vercel, but I like your approach 😉

Comment (chidubemokeke) [reply website/pages/en/cookbook/enums.mdx:5]: Awesome point!

**Review by alinobrasil — APPROVED** [review]
Looks great to me!

**Review by chidubemokeke — APPROVED** [review]

File: website/pages/en/cookbook/_meta.js [hunk 7]
```diff
@@ -11,4 +11,5 @@ export default {
   'immutable-entities-bytes-as-ids': 'Subgraph Best Practice 3: Using Immutable Entities and Bytes as IDs',
   'avoid-eth-calls': 'Subgraph Best Practice 4: Avoid eth_calls',
   'transfer-to-the-graph': '',
+  enums: 'Categorizing NFT Marketplaces Using Enums',
```
Comment (benface) [inline website/pages/en/cookbook/_meta.js:14]: @idalithb – Why customize the page title in the nav to make it so similar (just changing "Categorize" to "Categorizing")?

Comment (idalithb) [inline website/pages/en/cookbook/_meta.js:14]:
```suggestion
  enums: '',
```
